src/Knp/Component/Pager/Exception/PageNumberOutOfRangeException.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+
+namespace Knp\Component\Pager\Exception;
+
+use Throwable;
+
+class PageNumberOutOfRangeException extends \OutOfBoundsException
```

Comment:
should extend OutOfRangeException instead

Reply:
I've chosen OutOfBoundsException for two reasons:
1. [OutOfRangeException](https://php.net/manual/en/class.outofrangeexception.php) represents errors that should be detected at compile time, which is not the case here. [OutOfBoundsException](https://php.net/manual/en/class.outofboundsexception.php) is for errors that cannot be detected at compile time, which is the case.
2. OutOfRangeException extends LogicException and generic LogicException is thrown in this function in other cases, so this will be a sort of BCB.

Would you like to have OutOfRangeException nevertheless?

Reply:
I would go with OutofRangeException for two reasons:
1) it's more semantically correct (we launch exception with "page out of range", not "page out of bounds")
2) it makes possible to catch all exception by catching LogicException

Reply:
done